viewing_party/party.py

```diff
@@ -1,23 +1,169 @@
 
+def get_watched_avg_rating(user_data):
+    total_ratings = []
+    if user_data["watched"] != []:
+        try:
+            for movie in user_data["watched"]:
+                total_ratings.append(movie['rating'])
+
+            number_of_ratings = len(total_ratings)
+            average_rating = sum(total_ratings)/int(number_of_ratings)
+        except ZeroDivisionError:
+            average_rating = 0
+    else:
+        average_rating = 0
+
+    return average_rating
 
+#genre is a string
+def get_most_watched_genre(user_data):
+
+    if user_data["watched"] != []:
+        try:
+
+            most_watched = {}
+            description = []
+            for movie in user_data["watched"]:
+                if movie['genre']:
+                    description.append(movie['genre'])
+            for genre in description:
+                if genre in most_watched:
+                    most_watched[genre] += 1
+                else:
+                    most_watched[genre] = 1
+            most_popular = max(most_watched.values())
+            for keys, values in most_watched.items():
+                if values == most_popular:
+                    return keys
+                return user_data
+        except None:
+            return  None
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def create_friends_dictionary(user_data):
+    friends_movies = {}
+    friends_movies[1] = user_data["friends"][0]
+    friends_movies[2] = user_data["friends"][1]
+    return friends_movies
+
+def get_unique_watched(user_data):
+    friends_movies = create_friends_dictionary(user_data)
+    users_unique_movies = []
+
+    for title in user_data["watched"]:
+        if title not in friends_movies[1]["watched"] and title not in friends_movies[2]["watched"]:
+            users_unique_movies.append(title)
+    return users_unique_movies
+
+def get_friends_unique_watched(user_data):
+    friends_unique_movies = []
+    friends_movies = create_friends_dictionary(user_data)
+    
+    for movie in friends_movies[1]["watched"]:
+        if movie not in user_data["watched"]:
+            if movie not in friends_unique_movies:
+                friends_unique_movies.append(movie)
+    for friend_movies in friends_movies[2]["watched"]:
+        if friend_movies not in user_data["watched"]:
+            if friend_movies not in friends_unique_movies:
+                friends_unique_movies.append(friend_movies)
+    return friends_unique_movies
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    recommendations = []
+    friends_movies = create_friends_dictionary(user_data)
+
+    for movie in friends_movies[1]["watched"]:
+        if movie not in user_data["watched"] and movie["host"] in user_data["subscriptions"]:

```

Comment:
These checks enforce the before required of the description, but how could we structure this code overall to not rely on there being 2 friends?